src/psyclone/psyGen.py

```diff
@@ -5636,6 +5669,67 @@ def gen_c_code(self, indent=0):
         return self._reference
 
 
+class UnaryOperation(Node):
+    '''
+    Node representing a UnaryOperation expression. As such it has one operand
+    as children 0, and an attribute with the operator type.
+
+    :param str operator: string representing the unary operator.
+    :param parent: the parent node of this UnaryOperation in the PSyIR.
+    :type parent: :py:class:`psyclone.psyGen.Node`
+    '''
+    def __init__(self, operator, parent=None):
+        super(UnaryOperation, self).__init__(parent=parent)
+        self._operator = operator
```

Comment:
If possible I think it would be good to check here that the supplied operator is something valid.

Reply:
I am aware of this, and a proper solution is needed for nemolite2D (it is one of the reasons I decided to break this PR in smaller pieces).

I just created issue #339 and I will add a TODO in this line as a temporary notice.

Reply:
Great.